pkg/endpoint/endpoint.go

```diff
@@ -1644,9 +1644,16 @@ type MetadataResolverCB func(ns, podName string) (pod *slim_corev1.Pod, _ []slim
 // either the first successful metadata resolution or when the endpoint is
 // removed.
 //
+// initialLabels contains the list of labels that are added into the endpoint.
```

Comment:
I'm still not super happy with this go doc comment. initialLabels will **replace** cause any existing labels to be replaced with the set of `initialLabels +  resolveMetadata()`. The fact that is says "added" sounds like it's only additive, but it's not, it's a replacement.